lib/api/core/plugins/pluginContext.js

```diff
@@ -277,6 +277,7 @@ function registerStrategy(Strategy, name, context, verify, options = {}) {
 
   try {
     this.passport.use(name, new Strategy(opts, verify.bind(context)));
+    this.pluginsManager.registeredStrategies.push(name);
```

Comment:
Note: we can here fix also issue #762 : add the strategy to passport only if there is no strategy with the same name yet, and throw an error otherwise.